File: Array/chenjiaming/19删除链表倒数第N个节点/19.java
    class Solution {
        public ListNode removeNthFromEnd(ListNode head, int n) {
            ListNode p,q;
            p=q=head;

            while(p!=null){
               if(n<0){
                   q = q.next;
               }
               n--;
               p = p.next;
            }
            if(n==0){
                return head.next;
            }
            q.next = q.next.next;
            return head;


        }
    }

File: Array/chenjiaming/19删除链表倒数第N个节点/19.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* removeNthFromEnd(struct ListNode* head, int n){
struct ListNode *p =head,*q = head;
while(p!=NULL){

    if(n<0){
        q = q->next;
    }
    n--;
    p = p->next;


}

if(n==0){
    return head->next;

}

q->next = q->next->next;

return head;

}